Reindex from Remote from a newer version to older version

Let's say for example we're doing Reindex from Remote from 7.x -> 6.x version. 

a). Do we support this? 
If not, then I think we should probably document it better. 

b). If we do support this then I think below issue needs to be fixed. 
I tried this from 6.6 cluster and this is the error I get - 

```
[2020-04-24T12:43:34,597][WARN ][r.suppressed             ] [m_FSlpj] path: /_reindex, params: {pretty=}
org.elasticsearch.ElasticsearchException: Error parsing the response, remote is likely not an Elasticsearch instance
	at org.elasticsearch.index.reindex.remote.RemoteScrollableHitSource$1RetryHelper$1.onSuccess(RemoteScrollableHitSource.java:195) ~[?:?]
	at org.elasticsearch.client.RestClient$FailureTrackingResponseListener.onSuccess(RestClient.java:845) ~[?:?]
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:545) ~[?:?]
	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:533) ~[?:?]
	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119) ~[?:?]
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177) ~[?:?]
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436) ~[?:?]
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326) ~[?:?]
	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265) ~[?:?]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81) ~[?:?]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39) ~[?:?]
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114) ~[?:?]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162) ~[?:?]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337) ~[?:?]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315) ~[?:?]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276) ~[?:?]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[?:?]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: org.elasticsearch.common.xcontent.XContentParseException: [1:185] [search_response] failed to parse field [hits]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:316) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:326) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:168) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.apply(ConstructingObjectParser.java:161) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.apply(ConstructingObjectParser.java:76) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.index.reindex.remote.RemoteScrollableHitSource$1RetryHelper$1.onSuccess(RemoteScrollableHitSource.java:191) ~[?:?]
	... 18 more
Caused by: org.elasticsearch.common.xcontent.XContentParseException: [1:185] [hits] total doesn't support values of type: START_OBJECT
	at org.elasticsearch.common.xcontent.ObjectParser$FieldParser.assertSupports(ObjectParser.java:395) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:167) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.AbstractObjectParser.lambda$declareObject$1(AbstractObjectParser.java:146) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.lambda$declareField$1(ObjectParser.java:213) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:314) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:326) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:168) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.apply(ConstructingObjectParser.java:161) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.apply(ConstructingObjectParser.java:76) ~[elasticsearch-x-content-6.6.2.jar:6.6.2]
	at org.elasticsearch.index.reindex.remote.RemoteScrollableHitSource$1RetryHelper$1.onSuccess(RemoteScrollableHitSource.java:191) ~[?:?]
	... 18 more
```

This is because starting 7.0 version hits.total is a object - https://www.elastic.co/guide/en/elasticsearch/reference/7.0/breaking-changes-7.0.html#hits-total-now-object-search-response

Good way to solve this would be to add `rest_total_hits_as_int=true` but Reindex does not support that parameter yet. 

CC @talevy @jimczi @nik9000

--- a/docs/reference/upgrade/reindex_upgrade.asciidoc
+++ b/docs/reference/upgrade/reindex_upgrade.asciidoc
@@ -115,6 +115,9 @@ indices from the previous major version to be upgraded to the
 current major version. Skipping a major version means that you must
 resolve any backward compatibility issues yourself.
 
+{es} does not support downgrades or forward compatibility across major versions.
+For example, you cannot reindex from a 7.x cluster into a 6.x cluster.
+
 ifdef::include-xpack[]
 If you use {ml-features} and you're migrating indices from a 6.5 or earlier
 cluster, the job and {dfeed} configuration information are not stored in an

--- a/docs/reference/upgrade/reindex_upgrade.asciidoc
+++ b/docs/reference/upgrade/reindex_upgrade.asciidoc
@@ -115,7 +115,7 @@ indices from the previous major version to be upgraded to the
 current major version. Skipping a major version means that you must
 resolve any backward compatibility issues yourself.
 
-{es} does not support downgrades or forward compatibility across major versions.
+{es} does not support forward compatibility across major versions.
 For example, you cannot reindex from a 7.x cluster into a 6.x cluster.
 
 ifdef::include-xpack[]

